### PR TITLE
Add sequence ID, capture, publish and recieve timestamp to PhotonPipelineResult

### LINKED ISSUE #1304: Add pi packet send time and frame id to each packet

### PR DESCRIPTION
**Pull request by mcm001**
Closes #1304

### CODE REVIEW

File: photon-core/src/main/java/org/photonvision/common/util/math/MathUtils.java [hunk 1]
```diff
@@ -67,6 +67,10 @@ public static long microsToNanos(long micros) {
         return micros * 1000;
     }
 
+    public static long nanosToMicros(long micros) {
```
Comment (DevonRD) [inline photon-core/src/main/java/org/photonvision/common/util/math/MathUtils.java:70]: I think technically the parameter here is nanos

**Comment by srimanachanta** [comment]
Would this be a good time to make that change that includes the pipeline name in the result. Not sure if that overhead is worth.

**Comment by mcm001** [comment]
Maybe just the index number is a good compromise?

**Comment by TheTripleV** [comment]
We also sent across the capture timestamp in system time for future syncing.

lgtm otherwise.

**Comment by gerth2** [comment]
Fine as is, comments:
--> Timestamp caveats of "sometimes NT4 did not sync" - needs some integration thought later this summer.
--> Will break python as-is - OK, since theoretically pybind updates get that actually fixed
--> Sequence ID has some functional overlap with heartbeat. Maybe consider obsoleting heartbeat, or doing something else to save a byte or two in the messages?

**Review by gerth2 — APPROVED** [review]

**Comment by mcm001** [comment]
Ah whoops I can fix python. heartbeatPublisher is a good fallback since it shows up directly in Glass instead of inside a message. And yeah we should have a larger NTP discussion, but this is a good first step

**Pull request merged**